rotcon/convolve.py

```diff
@@ -114,22 +132,30 @@ def __init__(self, *args, **kwargs):
             rotamer1nitrogen = rotamersSite1.select_atoms("name N1")
             rotamer2nitrogen = rotamersSite2.select_atoms("name N1")
 
-            # loop over all the rotamers on the first site
-            for rotamer1 in rotamersSite1.trajectory:
-                if not rotamer1_clash[rotamer1.frame]:
-                    # loop over all the rotamers on the second site
-                    for rotamer2 in rotamersSite2.trajectory:
-                        if not rotamer2_clash[rotamer2.frame]:
-                            # measure and record the distance
-                            (a, b, distance) = MDAnalysis.analysis.distances.dist(rotamer1nitrogen, rotamer2nitrogen)
-                            distances.append(distance[0])
-                            # create the weights list
-                            try:
-                                weight = self.lib.weights[rotamer1.frame] * self.lib.weights[rotamer2.frame]
-                            except IndexError:
-                                logger.error("oppps: no weights for rotamer 1 #{0} - "
-                                             "rotamer 2 #{1}".format(rotamer1.frame, rotamer2.frame))
-                            weights.append(weight)
+            # define the atoms to measure the distances between
+            rotamer1All = rotamersSite1.select_atoms("all")
+            rotamer2All = rotamersSite2.select_atoms("all")
+            
+            with MDAnalysis.Writer("{}".format(tmptrjNoClashes[0]), rotamer1All.n_atoms) as S1:
```

Comment:
Writing out the additional files should be optional. In production mode you might not want to produce the additional files and have the additional overhead,